### PR TITLE
Wijax widgets

### PR DESCRIPTION
**Pull request by misterbisson**
### Purpose

Improve page load performance by delaying facet queries (and Scriblio Authority lookups on them) until requested via wijax.
### Standards improvements:
- whitespace cleanup
- remove `extract()` per https://github.com/GigaOM/legacy-pro/issues/2267

Review with [`?w=1`](https://github.com/misterbisson/scriblio/pull/37/files?w=1) for better health.

### CODE REVIEW

File: plugin/class-facets.php [hunk 1]
```diff
@@ -284,9 +284,18 @@ public function get_queryterms( $facet , $term , $additive = -1 )
...
 	{
-		$vars = apply_filters( 'scriblio_permalink_terms', $this->get_queryterms( $facet , $term , $additive ) );
+
+		if ( isset( $facet, $term ) )
+		{
+			$vars = $this->get_queryterms( $facet , $term , $additive );
+		}
+		else
+		{
+			$vars = clone $this->selected_facets;
```
Comment (misterbisson) [inline plugin/class-facets.php:296]: This is what allows getting the permalink to the current query. This line is copied from `get_queryterms()` above, where all currently selected terms are looked up and added to the new query.

File: plugin/widgets.php [hunk 2]
```diff
@@ -85,56 +81,55 @@ function widget( $args, $instance )
 			}//end if
 
 			// and now we wrap it all up for echo later
-			$content = scriblio()->facets()->generate_tag_cloud( $facet_list , $display_options );
+			$content = scriblio()->facets()->generate_tag_cloud( $facet_list, $display_options );
 		}
 		else
 		{
+			$wijax_source = trailingslashit( untrailingslashit( scriblio()->facets()->permalink() ) . '/wijax/' . bcms_wijax()->encoded_name( $this->id ) );
```
Comment (misterbisson) [inline plugin/widgets.php:88]: Note the use of the updated `permalink()` method here in place of `home_url()` that was causing the results in the widget to reflect global term usage, not the results of the specific query the user was looking at.

File: plugin/widgets.php [hunk 3]
```diff
@@ -29,17 +29,13 @@ function wijax_actions( $actions )
...
-		$title = apply_filters( 'widget_title' , empty( $instance['title'] ) ? '' : $instance['title'] );
-		$orderby = ( in_array( $instance['orderby'], array( 'count', 'name', 'custom' )) ? $instance['orderby'] : 'name' );
-		$order = ( in_array( $instance['order'], array( 'ASC', 'DESC' )) ? $instance['order'] : 'ASC' );
+		$title = apply_filters( 'widget_title', empty( $instance['title'] ) ? '' : $instance['title'] );
+		$orderby = ( in_array( $instance['orderby'], array( 'count', 'name', 'custom' ) ) ? $instance['orderby'] : 'name' );
+		$order = ( in_array( $instance['order'], array( 'ASC', 'DESC' ) ) ? $instance['order'] : 'ASC' );
 
 		// wijax requests get the whole thing
-		if( TRUE || is_wijax() )
+		if( ! function_exists( 'is_wijax' ) || is_wijax() )
```
Comment (misterbisson) [inline plugin/widgets.php:37]: I'd started work on a Wijax implementation ages ago, but didn't resolve a significant bug and just switched the feature off in this conditional.

File: plugin/widgets.php [hunk 4]
```diff
@@ -29,17 +29,13 @@ function wijax_actions( $actions )
...
 	{
-		global $mywijax;
-		extract( $args );
-
-		$title = apply_filters( 'widget_title' , empty( $instance['title'] ) ? '' : $instance['title'] );
-		$orderby = ( in_array( $instance['orderby'], array( 'count', 'name', 'custom' )) ? $instance['orderby'] : 'name' );
-		$order = ( in_array( $instance['order'], array( 'ASC', 'DESC' )) ? $instance['order'] : 'ASC' );
+		$title = apply_filters( 'widget_title', empty( $instance['title'] ) ? '' : $instance['title'] );
+		$orderby = ( in_array( $instance['orderby'], array( 'count', 'name', 'custom' ) ) ? $instance['orderby'] : 'name' );
+		$order = ( in_array( $instance['order'], array( 'ASC', 'DESC' ) ) ? $instance['order'] : 'ASC' );
```
Comment (borkweb) [inline plugin/widgets.php:34]: Both of these ternaries have extra parens.

File: plugin/widgets.php [hunk 5]
```diff
@@ -156,33 +151,33 @@ function form( $instance )
...
-			<label for="<?php echo $this->get_field_id('orderby'); ?>"><?php _e( 'Order By:' ); ?></label>
-			<select name="<?php echo $this->get_field_name('orderby'); ?>" id="<?php echo $this->get_field_id('orderby'); ?>" class="widefat">
-				<option value="count" <?php selected( $instance['orderby'], 'count' ); ?>><?php _e('Count'); ?></option>
-				<option value="name" <?php selected( $instance['orderby'], 'name' ); ?>><?php _e('Name'); ?></option>
-				<!-- <option value="custom" <?php selected( $instance['orderby'], 'custom' ); ?>><?php _e('Custom (see below)'); ?></option> -->
+			<label for="<?php echo $this->get_field_id( 'orderby' ); ?>"><?php _e( 'Order By:' ); ?></label>
+			<select name="<?php echo $this->get_field_name( 'orderby' ); ?>" id="<?php echo $this->get_field_id( 'orderby' ); ?>" class="widefat">
+				<option value="count" <?php selected( $instance['orderby'], 'count' ); ?>><?php _e( 'Count' ); ?></option>
+				<option value="name" <?php selected( $instance['orderby'], 'name' ); ?>><?php _e( 'Name' ); ?></option>
+				<!-- <option value="custom" <?php selected( $instance['orderby'], 'custom' ); ?>><?php _e( 'Custom (see below)' ); ?></option> -->
```
Comment (borkweb) [inline plugin/widgets.php:180]: Should this value be removed?

**Pull request merged**